Mage.Tests/src/test/java/org/mage/test/utils/ManaOptionsTest.java

```diff
@@ -603,13 +575,56 @@ public void testSungrassPrairie3() {
         execute();
 
         ManaOptions manaOptions = playerA.getAvailableManaTest(currentGame);
-        assertDuplicatedManaOptions(manaOptions);
 
-        Assert.assertEquals("mana variations don't fit", 4, manaOptions.size());
+        //Assert.assertEquals("mana variations don't fit", 4, manaOptions.size());
         assertManaOptions("{U}{U}", manaOptions);
         assertManaOptions("{G}{G}{W}", manaOptions);
         assertManaOptions("{G}{W}{U}", manaOptions);
         assertManaOptions("{G}{W}{W}", manaOptions);
     }
 
+    /**
+     * This is a stress test for the implementation of ManaUtils.
+     * Based on the bug from: https://github.com/magefree/mage/issues/7710
+     */
+    @Test
+    public void testCascadingCataracts() {
+        addCard(Zone.BATTLEFIELD, playerA, "Plains", 5);
+        addCard(Zone.BATTLEFIELD, playerA, "Island", 5);
+        addCard(Zone.BATTLEFIELD, playerA, "Swamp", 5);
+        addCard(Zone.BATTLEFIELD, playerA, "Mountain", 5);
+        addCard(Zone.BATTLEFIELD, playerA, "Forest", 5);
+        addCard(Zone.BATTLEFIELD, playerA, "Desert", 5);
+        addCard(Zone.BATTLEFIELD, playerA, "Cascading Cataracts", 3);
+
+        setStopAt(1, PhaseStep.END_TURN);
+        execute();
+        assertAllCommandsUsed();
+
+        ManaOptions manaOptions = playerA.getAvailableManaTest(currentGame);
+        Assert.assertEquals("mana variations don't match", 6902, manaOptions.size());
```

Comment:
It'd be helpful to add a comment block here explaining the calculation that brings the number of potential mana combinations to 6,902.